### PR TITLE
Autofocus field working on cards

### PR DESCRIPTION
**Pull request by atm-greg**
# FIX autofocus on cards
Make autofocus working on cards using configuration of focused fields

### CODE REVIEW

File: htdocs/core/lib/functions.lib.php [hunk 1]
```diff
@@ -9591,7 +9591,7 @@ function printCommonFooter($zone = 'private')
 			}
 
 			// Management of focus and mandatory for fields
-			if ($action == 'create' || $action == 'edit' || (empty($action) && (preg_match('/new\.php/', $_SERVER["PHP_SELF"])))) {
+			if ($action == 'create' || $action == 'edit' || (empty($action) && (preg_match('/new\.php/', $_SERVER["PHP_SELF"]))) || ((empty($action) || $action == 'addline') && (preg_match('/card\.php/', $_SERVER["PHP_SELF"])))) {
```
Comment (eldy) [inline htdocs/core/lib/functions.lib.php:9594]: The focus on card should be already managed by the  $action == 'create' || $action == 'edit' 
Adding a case   empty($action)   may result in focus forced in a lot of case we don't need.

Comment (atm-greg) [reply htdocs/core/lib/functions.lib.php:9594]: This is for lines form. We need this to focus on search_idprod.
When you are on a draft propal, the action is empty and after adding a line, the action is 'addline'.
Have you some suggestion to make it another way

Comment (eldy) [reply htdocs/core/lib/functions.lib.php:9594]: May be we can add 'addline'  ?
`if ($action == 'create' || $action == 'edit' || $action == 'addline') {`

Comment (eldy) [reply htdocs/core/lib/functions.lib.php:9594]: No better idea.
Lets do like that and we will see side effects.

**Pull request merged**